src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/BinaryBitwiseInstruction.t4

```diff
@@ -0,0 +1,86 @@
+﻿<#@ assembly name="System.Core" #>
+<#@ import namespace="System.Collections.Generic" #>
+<#@ import namespace="System.IO" #>
+<#@ import namespace="System.Linq" #>
+<#@ import namespace="System.Linq.Expressions" #>
+<#@ import namespace="System.Text" #>
+<#@ include file="Helpers.t4" #>
+<#
+string cachedInstructions = string.Join(", ", operands.Select(o => "s_" + toBcl(o)));
+#>
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Do not edit this file manually. This file is auto-generated by <#=Path.GetFileName(this.Host.TemplateFile)#>.
+
+using System.Dynamic.Utils;
+using System.Reflection;
+
+namespace System.Linq.Expressions.Interpreter
+{
+    internal abstract partial class <#=instruction#>Instruction : Instruction
+    {
+        private static Instruction <#=cachedInstructions#>;
+
+        public override int ConsumedStack => 2;
+        public override int ProducedStack => 1;
+        public override string InstructionName => "<#=instruction#>";
+
+        private <#=instruction#>Instruction() { }
+
+<#
+foreach (Type type in operands.Except(new[] { typeof(bool) }))
+{
+    string typeName = toBcl(type);
+    string typeAlias = toCSharp(type);
+
+    string op = "(" + typeAlias + ")left " + infix + " (" + typeAlias + ")right";
+
+    if (getBinaryResultType(infix, type, typeof(int)) != type)
+    {
+        op = "(" + typeAlias + ")(" + op + ")";
+    }
+#>
+        private sealed class <#=instruction#><#=typeName#> : <#=instruction#>Instruction
+        {
+            public override int Run(InterpretedFrame frame)
+            {
+                object left = frame.Pop();
+                object right = frame.Pop();
+                if (left == null || right == null)
+                {
+                    frame.Push(null);
+                    return 1;
+                }
+                frame.Push(<#=op#>);
+                return 1;
+            }
+        }
+
+<#
+}
+#>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA1502:AvoidExcessiveComplexity")]
+        public static Instruction Create(Type type)
+        {
+            // Boxed enums can be unboxed as their underlying types:
+            Type underlyingType = type.GetTypeInfo().IsEnum ? Enum.GetUnderlyingType(type) : type.GetNonNullableType();
```

Comment:
This is incorrect (but copied from what is incorrect in the currently code) and only doesn't error because it's also unreachable. Firstly, if we allowed `enum` types here, then it should be:

```C#
Type underlyingType = type.GetTypeInfo().IsEnum ? Enum.GetUnderlyingType(type.GetNonNullableType()) : type.GetNonNullableType();
```

Otherwise a nullable form of an enum type would `ArgumentException` here.

Also, the type of the result would be wrong, so `TypeIs(And(Constant(BindingFlags.Public), Constant(BindingFlags.Instance)), typeof(BindingFlags))` would always result in false when it should always result in true.

Finally since the only thing we do with `underlyingType` is get the typecode, which would be the same anyway, it isn't even necessary for that reason.

I'll be proposing soon that we actually allow that expression, but in the meantime should we replace this with just. `switch(type.GetNonNullableType().GetTypeCode())`?